### PR TITLE
More updates for future-proofing

### PR DESCRIPTION
**Pull request by nauaneed**

### CODE REVIEW

**Comment by nauaneed** [comment]
Pysph performance is not tested.

File: cyarray/carray.pyx.mako [hunk 1]
```diff
@@ -186,17 +192,17 @@ cdef class BaseArray:
...
         """Return the c data type of this array.
         """
-        raise NotImplementedError, 'BaseArray::get_c_type'
+        raise NotImplementedError('BaseArray::get_c_type')
 
     cpdef reserve(self, long size):
         """Resizes the internal data to required size.
         """
-        raise NotImplementedError, 'BaseArray::reserve'
+        raise NotImplementedError(BaseArray::reserve')
```
Comment (prabhuramachandran) [inline cyarray/carray.pyx.mako:200]: Please fix this error and the other one.

**Review by prabhuramachandran — CHANGES_REQUESTED** [review]
Please check if this is still compatible with Numpy versions < 2 and fix the silly errors.

**Review by prabhuramachandran — APPROVED** [review]
LGTM, thanks.

**Pull request merged**